File: mambo-protocol/src/main/scala/org/mambo/protocol/types/game/shortcut/ShortcutEmote.scala





















// Generated on 05/03/2013 10:35:08
// Scala version by Blackrush
package org.mambo.protocol.types

import org.mambo.core.io._
import org.mambo.protocol.{Type => BaseType, _}
import org.mambo.protocol.enums._

trait ShortcutEmote extends Shortcut {
  override val typeId = 389

  protected trait Type extends Shortcut.Type with Serializable {
    override val typeId = ShortcutEmote.typeId

    val emoteId: Byte

    override def serialize(buf: Buffer) {
      super.serialize(buf)

      buf.writeByte(emoteId)
    }

  }

  protected def create(slot: Int, emoteId: Byte): Target = throw new IllegalAccessException

  override def deserialize(buf: Buffer) = {
    val slot = buf.readInt()
    val emoteId = buf.readByte()

    create(slot, emoteId)
  }

}

object ShortcutEmote extends ShortcutEmote {
  case class TypeImpl(slot: Int, emoteId: Byte) extends Type
  override def create(slot: Int, emoteId: Byte) = TypeImpl(slot, emoteId)

  def unapply(msg: TypeImpl) = TypeImpl.unapply(msg)
  def apply(slot: Int, emoteId: Byte) = create(slot, emoteId)

  type Target = TypeImpl

}

File: mambo-protocol/src/main/java/org/mambo/protocol/types/ShortcutEmote.java


// Generated on 05/08/2013 19:38:03
package org.mambo.protocol.types;

import java.util.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class ShortcutEmote extends Shortcut {
    public static final short TYPE_ID = 389;
    
    @Override
    public short getTypeId() {
        return TYPE_ID;
    }
    
    public byte emoteId;
    
    public ShortcutEmote() { }
    
    public ShortcutEmote(int slot, byte emoteId) {
        super(slot);
        this.emoteId = emoteId;
    }
    
    @Override
    public void serialize(Buffer buf) {
        super.serialize(buf);
        buf.writeByte(emoteId);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        super.deserialize(buf);
        emoteId = buf.readByte();
        if (emoteId < 0)
            throw new RuntimeException("Forbidden value on emoteId = " + emoteId + ", it doesn't respect the following condition : emoteId < 0");
    }
    
}
